src/stores/main/middleware.ts

```diff
@@ -1,14 +1,18 @@
+import { FlashMessage, messageTTL, messageType } from 'lib/flashMessage';
 import { isJwtExpired } from 'lib/helpers';
 import { IOAuth2Provider } from 'lib/OAuth2/OAuth2';
 import { Middleware } from 'redux';
-import { logout } from 'stores/main/actions';
 import { getLoggedInUser } from 'stores/main/selectors';
 
 import { LoggedInUserTypes } from './types';
 
 export function mainAuthMiddleware(auth: IOAuth2Provider): Middleware {
   // eslint-disable-next-line consistent-return
   return (store) => (next) => async (action) => {
+    if (action.type?.startsWith('LOGOUT_')) {
+      return next(action);
+    }
```

Comment:
If we logged the user out after failing to renew the token, then the logout actions would be dispatched. If we don't ignore them, then we'll try to renew the user again, which would fail, and we'll dispatch the logout actions again, resulting in a memory leak.